src/offer/offerService.js

```diff
@@ -1,12 +1,38 @@
+import { FailedCreatingError } from '../errors/errorTypes/failedCreatingError.js';
 import { FailedToDeleteError } from '../errors/ErrorTypes/failedToDeleteError.js';
-import { InvalidDataError } from '../errors/ErrorTypes/invalidDataError.js';
-import { FAILED_DELETING, INVALID_DATA, OFFERS } from '../utils/textConstants.js';
+import { InvalidDataError } from '../errors/errorTypes/invalidDataError.js';
+import { FAILED_CREATE, FAILED_DELETING, INSUFFICIENT_QUANTITY, INVALID_DATA, OFFERS } from '../utils/textConstants.js';
 
 export class OfferService {
-  constructor ({ offerRepository }) {
+  constructor ({ offerRepository, inventoryRepository }) {
     this.offerRepository = offerRepository;
+    this.inventoryRepository = inventoryRepository;
   }
 
+  create = async (userName, offer, request, id) => {
+    if ((!userName) || (offer.length === 0) || (request.length === 0) || (!id)) {
+      throw new InvalidDataError(INVALID_DATA, OFFERS);
+    }
+    for (const item of offer) {
+      const { name, Quantity } = item;
+      const actualQuantity = await this.inventoryRepository.getQuantity(userName, name);
+      if (Quantity > actualQuantity) {
+        throw new InvalidDataError(INSUFFICIENT_QUANTITY, OFFERS);
+      }
+    }
+    for (const item of offer) {
+      const { name, Quantity } = item;
+      const isDelete = await this.inventoryRepository.removeItemToUser(userName, name, Quantity);
```

Comment:
Maybe at this point it is not the best idea to deconstruct item, it is more readable to do item.prop